src/config.rs

```diff
@@ -373,6 +394,12 @@ fn merge_configuration_files(
     output.files = first_package.files;
     output.variables = first_package.variables;
 
+    if let DefaultTargetType::Symbolic = global.settings.default_target_type {
+        output.files = transform_file_targets(output.files, FileTargetTransform::Symbolic);
+    } else if let DefaultTargetType::Template = global.settings.default_target_type {
+        output.files = transform_file_targets(output.files, FileTargetTransform::Template);
+    }
+
```

Comment:
IMO you overthought this a little with the FileTargetTransform. How about

```rust
for value in files.values_mut() {
    *value = match *value {
        FileTarget::Automatic(target) if global.settings.default_target_type == DefaultTargetType::Symbolic => {
            FileTarget::Symbolic(SymbolicTarget::from(target))
        },
        FileTarget::Automatic(target) if global.settings.default_target_type == DefaultTargetType::Template => {
            FileTarget::ComplexTemplate(TemplateTarget::from(target))
        },
        other => other,
    }
}
```
Or maybe
```rust
for value in files.values_mut() {
    if let FileTarget::Automatic(target) = *value {
        match global.settings.default_target_type {
            DefaultTargetType::Symbolic => {
                *value = FileTarget::Symbolic(SymbolicTarget::from(target));
            },
            DefaultTargetType::Template => {
                *value = FileTarget::ComplexTemplate(TemplateTarget::from(target));
            },
            _ => {},
        }
    }
}
```
Both of those might run into borrowing issues so you might have to use `.into_iter().map` 🙃 
The block could be extracted into a function that takes `default_target_type` if you think it would be cleaner

Reply:
yeah, I definitely agree re overthinking it. I like either of those options much better and will be working on implementing one. thanks.